goserver/session/manager.go

```diff
@@ -0,0 +1,59 @@
+package session
+
+import (
+	"fmt"
+	"time"
+)
+
+// Manager manages sessions for the server.
+type Manager interface {
```

Comment:
The way this is structured doesn't lead to much useful integration testing of this package. All the InMemorySessionManager code is purely for unit tests and won't share much code with the real implementation once it's added. It also requires the use of an interface which we generally try to avoid in Go as it makes code less readable (e.g. you also have to figure out which real impl is being used when reading code). So I would suggest the following:

```
type Store interface {
  Store(key string, val []byte) error
  Retrieve(key string) ([]byte, error)
}

type Manager struct {}

func NewManager(ss Store) {}
```
Then you can add a `sessiontest` package that provides an in-memory impl of `SessionStore`. This way you'll get more unit test coverage of the logic within SessionManager, keep test logic within test packages and SessionManager no longer needs to be an interface.

Reply:
Sounds great!